core/store/orm/orm.go

```diff
@@ -344,6 +345,12 @@ func (orm *ORM) FindExternalInitiator(eia *models.ExternalInitiatorAuthenticatio
 	return initiator, nil
 }
 
+// FindExternalInitiatorByName finds an external initiator given an authentication request
+func (orm *ORM) FindExternalInitiatorByName(eia *models.ExternalInitiatorAuthentication, name string) (models.ExternalInitiator, error) {
+	var exi models.ExternalInitiator
+	return exi, orm.DB.First(&exi, "access_key = ? AND name = ?", eia.AccessKey, name).Error
```

Comment:
I'd see name as a friendly descriptor not something to look up with, especially since we have an access key. Advantage of this is the name can be changed to suit your needs, can have punctuation etc. without affecting lookup.

Reply:
This added in support of validating whether a external initiator name already exists.

My assumption re: access key was it was bound to the user, not the external initiator.  Is it specific only to the external initiator?